presto-main/src/main/java/com/facebook/presto/operator/SortedPositionLinks.java

```diff
@@ -52,22 +55,35 @@
 public final class SortedPositionLinks
         implements PositionLinks
 {
-    public static class Builder implements PositionLinks.Builder
+    public static class Builder
+            implements PositionLinks.Builder
     {
         private final Int2ObjectMap<IntArrayList> positionLinks;
         private final int size;
         private final IntComparator comparator;
+        private final PagesHashStrategy pagesHashStrategy;
+        private final LongArrayList addresses;
 
-        public Builder(int size, IntComparator comparator)
+        public Builder(int size, IntComparator comparator, PagesHashStrategy pagesHashStrategy, LongArrayList addresses)
         {
-            this.comparator = comparator;
             this.size = size;
+            this.comparator = comparator;
+            this.pagesHashStrategy = pagesHashStrategy;
+            this.addresses = addresses;
             positionLinks = new Int2ObjectOpenHashMap<>();
         }
 
         @Override
         public int link(int from, int to)
         {
+            if (isNull(from)) {
+                return to;
+            }
+
+            if (isNull(to)) {
```

Comment:
what if both are nulls or all of the values are null? Do you have tests for that?